apps/src/javalab/JavabuilderConnection.js

```diff
@@ -104,6 +104,19 @@ export default class JavabuilderConnection {
 
   // Send a message across the websocket connection to Javabuilder
   sendMessage(message) {
-    this.socket.send(message);
+    if (!this.socket) {
```

Comment:
nit: can turn this function (and `closeConnection`) into just `if(this.socket)` (no else required)

Reply:
🙌